.github/workflows/ci.yaml

```
@@ -0,0 +1,59 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - dev
+      - main
+
+  pull_request:
+    branches:
+      - dev
+      - main
+
+concurrency:
+  group: ci-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎
+        uses: actions/checkout@v3
+
+      - name: Install 👨🏻‍💻
+        run: npm i
+
+      - name: Lint 👀
+        run: npm run lint
+
+      - name: Prettier 🪞
+        run: npm run format:fix
+
+  type_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout 🛎
+        uses: actions/checkout@v3
+
+      - name: Install 👨🏻‍💻
+        run: npm i
+
+      - name: Type check 🔤
+        run: npm run type-check
+
+  test:
+    runs-on: ubuntu-latest
+
+    timeout-minutes: 5
+
+    steps:
```

The indents look redundant. Let's write beautifully even in configuration files)

```suggestion
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
```